core/src/test/java/io/questdb/griffin/SqlCodeGeneratorTest.java

```diff
@@ -6401,6 +6391,443 @@ public void testVirtualColumnsSansTimestamp() throws Exception {
+            executeInsert("insert into tab values ('d2', 'c1', 201.4, '2021-10-05T14:31:35.878Z')");
+            executeInsert("insert into tab values ('d2', 'c2', 401.1, '2021-10-06T11:31:35.878Z')");
+            executeInsert("insert into tab values ('d2', 'c1', 401.2, '2021-10-06T12:31:35.878Z')");
+            executeInsert("insert into tab values ('d2', 'c1', 111.7, '2021-10-06T15:31:35.878Z')");
+
+            assertSql(
+                    "tab latest by id, name where id = 'd1'",
+                    "id\tname\tvalue\tts\n" +
+                            "d1\tc1\t101.4\t2021-10-05T14:31:35.878000Z\n" +
+                            "d1\tc2\t102.5\t2021-10-05T15:31:35.878000Z\n");
+            assertSql(
+                    "tab latest by id, name where id != 'd2' and value < 102.5",
+                    "id\tname\tvalue\tts\n" +
+                            "d1\tc1\t101.4\t2021-10-05T14:31:35.878000Z\n" +
+                            "d1\tc2\t102.4\t2021-10-05T14:31:35.878000Z\n");
+            assertSql(
+                    "tab latest by id, name where name = 'c1'",
+                    "id\tname\tvalue\tts\n" +
+                            "d1\tc1\t101.4\t2021-10-05T14:31:35.878000Z\n" +
+                            "d2\tc1\t111.7\t2021-10-06T15:31:35.878000Z\n");
+            assertSql(
+                    "tab latest by id, name where name != 'c2' and value <= 111.7",
+                    "id\tname\tvalue\tts\n" +
+                            "d1\tc1\t101.4\t2021-10-05T14:31:35.878000Z\n" +
+                            "d2\tc1\t111.7\t2021-10-06T15:31:35.878000Z\n");
+            // TODO: broken 2,4,5,7
+            assertSql(
+                    "tab latest by id where name = 'c2'",
+                    "id\tname\tvalue\tts\n" +
+                            "d1\tc2\t102.5\t2021-10-05T15:31:35.878000Z\n" +
+                            "d2\tc2\t401.1\t2021-10-06T11:31:35.878000Z\n");
+            // TODO: broken 3,4,5,6
+            assertSql(
+                    "tab latest by name where id = 'd1'",
+                    "id\tname\tvalue\tts\n" +
+                            "d1\tc1\t101.4\t2021-10-05T14:31:35.878000Z\n" +
+                            "d1\tc2\t102.5\t2021-10-05T15:31:35.878000Z\n");
+            assertSql(
+                    "tab latest by name where id = 'd2'",
+                    "id\tname\tvalue\tts\n" +
+                            "d2\tc2\t401.1\t2021-10-06T11:31:35.878000Z\n" +
+                            "d2\tc1\t111.7\t2021-10-06T15:31:35.878000Z\n");
+            assertSql(
+                    "tab latest by name where id != 'd1'",
+                    "id\tname\tvalue\tts\n" +
+                            "d2\tc2\t401.1\t2021-10-06T11:31:35.878000Z\n" +
+                            "d2\tc1\t111.7\t2021-10-06T15:31:35.878000Z\n");
+        });
+    }
+
+    @Ignore("result order is currently dependent on stability of sorting method")
```

Comment:
What is this test testing? Does it simply inserts and queries back everything? If yes I don't see much value keeping it ignored

Reply:
The order of insert needs to be preserved, even if the timestamps are the exact same down to the microsecond level, they are inserted/committed one row at the time. We have to be able to say that our selects are stable and preserve some order, regardless of whether it is FIFO or LIFO. The test is there as a reminder, it will need to be fixed, but not in this PR